components/org.wso2.carbon.appmgt.rest.api.publisher/src/main/java/org/wso2/carbon/appmgt/rest/api/publisher/impl/AdministrationApiServiceImpl.java

```diff
@@ -23,44 +23,230 @@
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            String ownerName = body.getName();
+            String ownerEmail = body.getEmail();
+            String ownerDescription = body.getDescription();
+            String ownerSite = body.getSite();
+            if (StringUtils.isEmpty(ownerName)) {
+                RestApiUtil.handleBadRequest("Business owner name cannot be null or empty.", log);
+            }
+            if (StringUtils.isEmpty(ownerEmail)) {
+                RestApiUtil.handleBadRequest("Business owner email cannot be null or empty.", log);
+            }
+            BusinessOwner businessOwner = new BusinessOwner();
+            businessOwner.setBusinessOwnerId(Integer.parseInt(businessOwnerId));
+            businessOwner.setBusinessOwnerName(ownerName.trim());
+            businessOwner.setBusinessOwnerEmail(ownerEmail.trim());
+            businessOwner.setBusinessOwnerDescription(ownerDescription);
+            businessOwner.setBusinessOwnerSite(ownerSite);
+            List<BusinessOwnerPropertiesDTO> businessOwnerPropertiesDTOList = body.getProperties();
+
+            List<BusinessOwnerProperty> businessOwnerPropertyList = new ArrayList<>();
+            for (BusinessOwnerPropertiesDTO businessOwnerPropertiesDTO : businessOwnerPropertiesDTOList) {
+                String propertyId = businessOwnerPropertiesDTO.getKey();
+                String propertyValue = businessOwnerPropertiesDTO.getValue();
+                Boolean isVisible = businessOwnerPropertiesDTO.getIsVisible();
+                if (StringUtils.isEmpty(propertyId) || StringUtils.isEmpty(propertyValue) || isVisible == null) {
+                    RestApiUtil.handleBadRequest("Business owner properties cannot be null or empty.", log);
+                }
+
+                BusinessOwnerProperty businessOwnerProperty = new BusinessOwnerProperty();
+                businessOwnerProperty.setPropertyId(propertyId);
+                businessOwnerProperty.setPropertyValue(propertyValue);
+                businessOwnerProperty.setShowingInStore(isVisible);
+                businessOwnerPropertyList.add(businessOwnerProperty);
+            }
+            businessOwner.setBusinessOwnerPropertiesList(businessOwnerPropertyList);
+            apiProvider.updateBusinessOwner(businessOwner);
+        } catch (AppManagementException e) {
+            String errorMessage = "Error while updating Business owner for business owner Id " + businessOwnerId;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        }
+        return Response.ok().build();
     }
 
     @Override
     public Response administrationBusinessownerBusinessOwnerIdDelete(String businessOwnerId, String ifMatch, String ifUnmodifiedSince) {
-        return null;
+        try {
+            APIProvider apiProvider = RestApiUtil.getLoggedInUserProvider();
+            //delete the business owner.
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/0d9719587a489bf2ef909c862c46bdbce01f1af0